### ISSUE TITLE
Fuji X-T2 - no real success as yet.  Please help.

### ISSUE THREAD

**Issue by ACMACMACM** [gphoto/gphoto2#340]
Hi Marcus,

Ive had a chance to test GPhoto2 with my Fuji X-T2 and unfortunately theres some weird things happening, but hopefully nothing that cant be fixed.

**Camera setup:**
My camera has a physical dial with 3 shooting modes: A (auto), B (bulb), T (timed), and also a way to shoot in either MS (mechanical shutter) or ES (electronic shutter), or MS&ES (both).

**Testing method:**
I tested image capture from the command line, and then if that worked, also in Ekos capture module and noted the results. 

**Saving the image:**
In all cases though, the image was only stored on the Pi and that was only when capturing in Ekos (after i specified the output folder within Ekos).
If there is a way using the console command to save to a specific folder then that may have worked, however i wasnt aware of the syntax to do that.

**Data Indicator Lamp:**
The whole time the Fuji was plugged in the 'indicator lamp' on it flashed orange and green, as if data transfer was taking place, even long after images had been captured. It actually never stopped.  
Even after I had turned power off to the camera (which didnt actually power off the camera) the indicator lamp went solid orange as if data transfer was even still taking place.  Only when i removed the USB cable did the camera finally completely turn off.

**PC Connection Modes:**
I tested all 3 camera tether modes on the Fuji X-T2 under 'PC Connection Mode' menu:
- USB Card Reader: This causes the ERROR:  'Sorry your camera does not support generic capture' when running from command line.  Ekos didnt work obviously.
- USB Tether Shooting AUTO: Saves only to the Pi, and thats when shot from within Ekos after specifying a save directory (as aforementioned).  Running the 'gphoto2 --capture-image' command line in Pi terminal did say 'downloading' and when the progress bar for that reached 100%, it said 'image saved to card', however it actually hadnt as far as I could tell. 
- USB Tether Shooting Fixed: Same as the above, however a warning message appears stating 'waiting for data transfer to complete, Press OK to cancel', which doesnt appear using the above PC Connection Mode.

**Results:**
(B)ulb mode ONLY works with the ES shutter and it only shoots 1s images (which is a Fuji limitation on the BULB/ES combo anyway).  However if MS shutter or MS&ES shutter is selected, BULB mode stops working altogether and wont capture an image.  (logfile attached below)
(A)utomatic mode works (MS, ES, both) but cannot control the shutter speed at all (as the camera controls that for what it considers to be an optimal exposure). This happens for all MS, ES, MS&ES shutters.
(T)imed mode works (MS, ES, both) in the following way:
- Pi Terminal command line: Takes the image and uses the shutter speed that was set on the camera prior to connecting it to the Pi.  Image not saved to the camera or the Pi (unless its in a default directory unknown to me on the Pi).
- Ekos capture module: Requested exposure time entered was 120 seconds, and a 1.3s image was taken, however you do need to wait the following ~118 seconds before the image is saved to the Pi directory (directory specified in Ekos).  Image is not saved to the camera.  The orange and green lights flash continuously.  I tested a number of different exposure times and noticed that 'requested' times vs 'actual' times (according to the metadata in Lightroom) are seen to be divisable by around 100, actually somewhere between 90-110.  (eg.  60s requested = 0.6s actual, 180s = 2.0s actual, 900s = 8.0s actual, 10s = 0.1s actual).  
I have attached the logfile for this also.

**Logfiles:**
[BULB Mode Mechanical Shutter not working](https://github.com/gphoto/gphoto2/files/4776138/BULB.Mode.Mechanical.Shutter.txt)
I also created a logfile for BULB Mode with Electronic Shutter (that shot the 1s image successfully) but its 220MB in size.  Let me know if you might want that for any comparison.
[KStars Timed mode out by a factor of 90 to 110](https://github.com/gphoto/gphoto2/files/4776410/log_21-50-32.Timed.mode.out.by.factor.of.90.to.110.txt)

**Additional info:**
Also, the 'Preset' time values on the Ekos tab:'Gphoto CCD' had very high & unusual looking values.

**Versions:**
GPhoto version: 2.5.23, libgphoto2: 2.5.24.1 (even though i did upgrade to 2.5.25 it still says 2.5.24.1), libgphoto2_port 0.12.0.  The X-T2 camera was in the mentioned above modes, connected to a Raspian Raspberry Pi 4B.

Hopefully, the BULB mode issue when set to MS / MS&ES can be fixed.
As an alternative, the TIMED mode might be a secondary solution but somehow have to fix the 'out by a factor' issue (in GPhoto or KStars/Ekos.. pls let me know if i should raise that there).

Many thanks,

**Comment by lfbmendes** [comment]
I'm also very interested in a solution about this issue. If you need a tester, please let me know.

**Comment by riccardolardi** [comment]
Just wondering if there‘s any update on this? Wondering if a X-T1 or X-T2 could be used for astrophotography with Ekos.

**Comment by sperho** [comment]
I have an X-T3 and an X-T4, but many of the OP's described issues (except for the factor of 100 error) apply to these cameras, too.  I would love to use my Fuji X cameras for astrophotography as well.  Is there any progress on debugging the fuji INDI driver?  Thanks.

**Comment by msmeissn** [comment]
I rented a Fuji XT-3 for the weekend, largely integrating other stuff.

Do you know a free (for download) windows app that allows bulb shooting where I could sniff traffic?

**Comment by riccardolardi** [comment]
@msmeissn hey I was trying to do te same. Are you going to use wireshark? I know there is a the official Fuji iOS app for wifi tethering

**Comment by sperho** [comment]
> I rented a Fuji XT-3 for the weekend, largely integrating other stuff.
> 
> Do you know a free (for download) windows app that allows bulb shooting where I could sniff traffic?

I replied to you or other post on this issue, but I just remembered that you can tether via the free Fuji X-Aquire Windows application downloadable from the Fuji site. I can test this evening to see it can do bulb exposures.

**Comment by riccardolardi** [comment]
There is also this https://github.com/hkr/fuji-cam-wifi-tool where soneone already started reverse engineering the fuji remote system

**Comment by sperho** [comment]
> > I rented a Fuji XT-3 for the weekend, largely integrating other stuff.
> > Do you know a free (for download) windows app that allows bulb shooting where I could sniff traffic?
> 
> I replied to you or other post on this issue, but I just remembered that you can tether via the free Fuji X-Aquire Windows application downloadable from the Fuji site. I can test this evening to see it can do bulb exposures.

I tested Fuji X-Acquire with the X-T3 and it can only read the state of the camera (including BULB), but it cannot release the shutter.  Its function as it relates to tethering is to 1) offer the user the ability to review the camera's settings on the computer as they are set on the camera dials and 2) an automatic transfer of files after the shutter is released on the camera itself.

**Comment by msmeissn** [comment]
yes, i tried xacquie too and it only reads from the camera, it cannot control it.
i tried also fuji webcam, but it dos not have remote capture mode (just live view streaming with some control)

I used USBpcap to capture .pcap files and read them with wireshark on linux

**Comment by msmeissn** [comment]
I used windows 10 ... not sure how to do it in Apple iOS

**Comment by lfbmendes** [comment]
> 
> 
> There is also this https://github.com/hkr/fuji-cam-wifi-tool where soneone already started reverse engineering the fuji remote system

This uses the protocol between smartphone and camera, and does not work for more demanding work (studio photo, astrophoto, etc). 

As soon as I can, I'll buy the Fuji tether plugin pro, that allows camera control from the PC. Too bad it costs 80 dollars.

**Comment by riccardolardi** [comment]
I would love to help if possible. What is your general approach, sniff the USB traffic to determine the protocol? I'm Linux savy and know my way around wireshark and I own a Fuji X-T10. Let me know how I can help

**Comment by msmeissn** [comment]
fuji-cam-wifi-tool ... i have the sources, but our current libgphoto2 implementation exceeds it already. Especially the capture code is not doing the special tricks we already do , and no bulb capture.

The xt10 does not support capture over USB I am afraid, it _might_ over WLAN.

**Comment by lfbmendes** [comment]
> 
> 
> The xt10 does not support capture over USB I am afraid, it _might_ over WLAN.

Sort of. It shoots, and it transfers. But in two different modes, one to shoot and another to receive. And the camera keeps thinking between modes. And it only transfer a low resolution JPG. I suggest focusing on the cameras that support tethering.
